src/main/controller/PromotionController.ts

```diff
@@ -168,11 +170,24 @@ export class PromotionController {
         const id = await IdValidation.schema.validateAsync(request.params.id, {
           abortEarly: false,
         });
-        const promotion = await transactionalEntityManager
+
+        const authenticatedUser = await transactionalEntityManager
+          .getCustomRepository(UserRepository)
+          .findByFirebaseId(response.locals.firebaseUserId);
+
+        const potentialPromotionUploadedByUser = await transactionalEntityManager
+          .getCustomRepository(PromotionRepository)
+          .findOne({ id, user: authenticatedUser });
```

Comment:
@ericrkuo , this is good way to find authorized user.  Could you change user deletion in the same way to make the implementation consistent between promotion and user?

Reply:
Ohh I see, kk will do good point